pkg/kube/controllers/quarkssecret/quarkssecret_controller.go

```diff
@@ -97,6 +102,64 @@ func AddQuarksSecret(ctx context.Context, config *config.Config, mgr manager.Man
 		return errors.Wrapf(err, "Watching quarks secrets failed in quarksSecret controller.")
 	}
 
+	// Watch for changes to user created secrets
+	p = predicate.Funcs{
+		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			n := e.ObjectNew.(*corev1.Secret)
+			o := e.ObjectOld.(*corev1.Secret)
+
+			shouldProcessReconcile := isUserCreatedSecret(n)
+			if reflect.DeepEqual(n.Data, o.Data) && reflect.DeepEqual(n.Labels, o.Labels) &&
+				reflect.DeepEqual(n.Annotations, o.Annotations) {
+				return false
+			}
+
+			return shouldProcessReconcile
+		},
+	}
+	err = c.Watch(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
+			secret := a.Object.(*corev1.Secret)
+
+			if skip.Reconciles(ctx, mgr.GetClient(), secret) {
+				return []reconcile.Request{}
+			}
+
+			reconciles, err := reference.GetReconciles(ctx, mgr.GetClient(), reference.ReconcileForQuarksSecret, secret, false)
```

Comment:
There has to be an easier way than to copy the `GetReconciles` package.

Reply:
Done